npm.d.ts

```diff
@@ -0,0 +1,439 @@
+*/
+export interface Dependency {
+	[packageName: string]: string;
+}
+
+export type PackageJson = {
+	/**
+	The name of the package.
+	*/
+	name?: string;
+
+	/**
+	Package version, parseable by [`node-semver`](https://github.com/npm/node-semver).
+	*/
+	version?: string;
+
+	/**
+	Package description, listed in `npm search`.
+	*/
+	description?: string;
+
+	/**
+	Keywords associated with package, listed in `npm search`.
+	*/
+	keywords?: string[];
+
+	/**
+	The URL to the package's homepage.
+	*/
+	homepage?: LiteralUnion<'.', string>;
+
+	/**
+	The URL to the package's issue tracker and/or the email address to which issues should be reported.
+	*/
+	bugs?: BugsLocation;
+
+	/**
+	The license for the package.
+	*/
+	license?: string;
+
+	/**
+	The licenses for the package.
+	*/
+	licenses?: {
+		type?: string;
+		url?: string;
+	}[];
+
+	author?: Person;
+
+	/**
+	A list of people who contributed to the package.
+	*/
+	contributors?: Person[];
+
+	/**
+	A list of people who maintain the package.
+	*/
+	maintainers?: Person[];
+
+	/**
+	The files included in the package.
+	*/
+	files?: string[];
+
+	/**
+	The module ID that is the primary entry point to the program.
+	*/
+	main?: string;
+
+	bin?:
+	| string
+	| {
+		[binary: string]: string;
+	};
+
+	/**
+	Location of the bundled TypeScript declaration file.
+	*/
+	types?: string;
+
+	/**
+	Location of the bundled TypeScript declaration file. Alias of `types`.
+	*/
+	typings?: string;
+
+	/**
+	Filenames to put in place for the `man` program to find.
+	*/
+	man?: string | string[];
+
+	directories?: DirectoryLocations;
+
+	/**
+	Location for the code repository.
+	*/
+	repository?:
+	| string
+	| {
+		type: string;
+		url: string;
+	};
+
+	/**
+	Script commands that are run at various times in the lifecycle of the package. The key is the lifecycle event, and the value is the command to run at that point.
+	*/
+	scripts?: ScriptsDefinition;
+
+	/**
+	Is used to set configuration parameters used in package scripts that persist across upgrades.
+	*/
+	config?: {
+		[configKey: string]: unknown;
+	};
+
+	/**
+	The dependencies of the package.
+	*/
+	dependencies?: Dependency;
+
+	/**
+	Additional tooling dependencies that are not required for the package to work. Usually test, build, or documentation tooling.
+	*/
+	devDependencies?: Dependency;
+
+	/**
+	Dependencies that are skipped if they fail to install.
+	*/
+	optionalDependencies?: Dependency;
+
+	/**
+	Dependencies that will usually be required by the package user directly or via another dependency.
+	*/
+	peerDependencies?: Dependency;
+
+	/**
+	Package names that are bundled when the package is published.
+	*/
+	bundledDependencies?: string[];
+
+	/**
+	Alias of `bundledDependencies`.
+	*/
+	bundleDependencies?: string[];
+
+	resolutions?: Dependency;
+
+	/**
+	Engines that this package runs on.
+	*/
+	engines?: {
+		[engineName: string]: string;
```

Comment:
This should at least define `node` and `npm` as engine. https://github.com/chalk/chalk/blob/cd5de7a2f6a7a04fbad4c75c0364a3efc6048ade/package.json#L8 In addition to the above.

https://docs.npmjs.com/files/package.json#engines